### PR TITLE
Soft delete projects

### LINKED ISSUE #187: [8] Soft delete projects & hard delete slugs & update select statements
Update database statements to soft delete everything except slugs (to avoid slug conflict down the road).

The docs have been updated and should now reflect what this change will look like.

refs #173

### PR DESCRIPTION
**Pull request by LyonesGamer**
Fixes #187.

### CODE REVIEW

File: src/projects.js [hunk 1]
```diff
@@ -549,32 +550,80 @@ module.exports = function(app) {
...
 
-        const err = errors.errorObjectNotFound('slug',
-        req.params.slug);
-        return res.status(err.status).send(err);
+        /*
+         * Once auth is provided on DELETE requests, compare user ID to ensure
+         * they have either 'project manager' rights on the project, or admin
+         * rights on the system.
+         */
+        knex('userroles').where({project: project.id, user: 0 /* userId */ })
```
Comment (athai) [inline src/projects.js]: Shouldn't you be selecting user based on 'user.id' and not explicitly state the id?

Comment (LyonesGamer) [inline src/projects.js]: Sorry, that was sort of pseudocode which should have been commented out. Fixed now.

**Comment by Kennric** [comment]
This looks pretty good - but  get another review. +1

**Comment by ElijahCaine** [comment]
Everything looks good to me. Tests pass and the changes made make sense. +1

**Pull request merged**